### PR TITLE
CLOUD-411 integrate OpenStack

### PR DESCRIPTION
**Pull request by keyki**

### CODE REVIEW

**Comment by keyki** [comment]
SQL scripts are in progress

File: src/main/java/com/sequenceiq/cloudbreak/controller/validation/OpenStackTemplateParam.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package com.sequenceiq.cloudbreak.controller.validation;
+
+import com.google.common.base.Optional;
+import com.sequenceiq.cloudbreak.domain.AzureVmType;
+
+public enum OpenStackTemplateParam implements TemplateParam {
+
+    PUBLIC_NET_ID("publicNetId", true, String.class, Optional.<String>absent()),
+    INSTANCE_TYPE("instanceType", true, AzureVmType.class, Optional.<String>absent());
```
Comment (doktoric) [inline src/main/java/com/sequenceiq/cloudbreak/controller/validation/OpenStackTemplateParam.java]: why do you use azurevmtype

File: src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/openstack/HeatTemplateBuilder.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        }
+        return volumes;
+    }
+
+    private String formatUserData(String userData) {
+        String[] lines = userData.split("\n");
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < lines.length; i++) {
+            // be aware of the OpenStack Heat template formatting
+            sb.append("            " + lines[i] + "\n");
```
Comment (doktoric) [inline src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/openstack/HeatTemplateBuilder.java]: what is this?

**Comment by doktoric** [comment]
please modify the qa-deploy script with the new image url and open a pull request in uluwatu.
After that I will merge it to the master

**Pull request merged**